pkg/generators/system/config/options.go

```diff
@@ -123,8 +123,8 @@ func NewOptions(spec saasv1alpha1.SystemSpec) Options {
 		BackendRedisURL:            &pod.ClearTextValue{Value: spec.Config.Backend.RedisDSN},
 		BackendRedisSentinelHosts:  &pod.ClearTextValue{Value: ""},
 		BackendRedisSentinelRole:   &pod.ClearTextValue{Value: ""},
-		ApicastBackendRootEndpoint: &pod.ClearTextValue{Value: spec.Config.Backend.InternalEndpoint},
-		BackendRoute:               &pod.ClearTextValue{Value: spec.Config.Backend.InternalEndpoint},
+		ApicastBackendRootEndpoint: &pod.ClearTextValue{Value: spec.Config.Backend.ExternalEndpoint},
```

Comment:
This is OK, [3scale operator](https://github.com/3scale/3scale-operator/blob/ede8838d156d1bf06a25391668f7b5bc18e4cc58/pkg/3scale/amp/component/system.go#LL208C129-L208C179) also points to the public endpoint (route). The change it, in fact, just to align, with on-prem 3scale operator.

The configs for SaaS do not use this env var, they rely on `BACKEND_PUBLIC_URL` instead.
(which means that maybe we could potentially drop `APICAST_BACKEND_ROOT_ENDPOINT`)